### PR TITLE
added additional auth check on get results

### PR DESCRIPTION
**Pull request by philipjusher**

### CODE REVIEW

File: src/ansys/conceptev/core/app.py [hunk 1]
```diff
@@ -337,6 +337,7 @@ def read_results(
     job_info: dict,
     calculate_units: bool = True,
     timeout: int = JOB_TIMEOUT,
+    msal_app: auth.PublicClientApplication | None = None,
```
Comment (RobKellyAnsys) [inline src/ansys/conceptev/core/app.py:340]: why is this an input? I can't see where we use it

**Review by RobKellyAnsys — CHANGES_REQUESTED** [review]

File: src/ansys/conceptev/core/app.py [hunk 2]
```diff
@@ -347,6 +348,10 @@ def read_results(
         return get_results(client, job_info, calculate_units)
     else:  # Job is still running
         monitor_job_progress(job_id, user_id, token, timeout)  # Wait for completion
+        if msal_app is None:
```
[the remark this answers is not in the document]
Comment (RobKellyAnsys) [reply src/ansys/conceptev/core/app.py:351]: yes but where does it get called with that input?

[the remark this answers is not in the document]
Comment (philipjusher) [reply src/ansys/conceptev/core/app.py:351]: oh thats up to Georg to add to the call. Its optional so if he doesn't make it that's fine. The simple_workflow file has an example. I think I changed it in there but might have left it to use the default.

[the remark this answers is not in the document]
Comment (RobKellyAnsys) [reply src/ansys/conceptev/core/app.py:351]: ah ok gotya

**Review by RobKellyAnsys — APPROVED** [review]

**Pull request merged**